test/auth/features/oidc/broker.go

```diff
@@ -42,13 +44,19 @@ func BrokerSendEventWithOIDCToken() *feature.Feature {
 	f.Setup("broker is addressable", broker.IsAddressable(brokerName))
 
 	// Install the sink
-	f.Setup("install sink", eventshub.Install(sink, eventshub.StartReceiver))
+	f.Setup("install sink", eventshub.Install(
+		sink,
+		eventshub.OIDCReceiverAudience(sinkAudience),
+		eventshub.StartReceiver))
 
 	// Install the trigger and Point the Trigger subscriber to the sink svc.
 	f.Setup("install trigger", trigger.Install(
 		triggerName,
 		brokerName,
-		trigger.WithSubscriber(service.AsKReference(sink), ""),
+		trigger.WithSubscriberFromDestination(&v1.Destination{
+			Ref:      service.AsKReference(sink),
+			Audience: &sinkAudience,
```

Comment:
Just curious why we need the sink-audience here? If we pass the audience to both subscriber and the sink, does that mean the token will be validated twice?

Here is my understanding:
At the Trigger:  When the trigger forwarding the event, ensures that the token it's using is intended for the specific sink. This is more about token generation and ensuring it's crafted with the correct audience in mind

At the Sink: The sink, upon receiving the event, will validate the token. It's a form of asserting that the event it received was indeed meant for it

Please correct me if this is wrong. @creydr

Reply:
When we pass the audience when we create the sink via `eventshub.OIDCReceiverAudience(sinkAudience)`, this tells the eventshub receiver, that it should expect a JWT for this audience (the eventshub receiver validates the JWT).
Since the eventhub receiver does not announce its audience in its status, the trigger reconciler has no chance to know which audience its subscriber (the eventhub receiver/sink) expects. Therefor we setup the triggers subscriber via `trigger.WithSubscriberFromDestination()`, which allows us to specify the audience too. (I could have updated the signature of `trigger.WithSubscriber()` too, to have an additional argument, but via `WithSubscriberFromDestination()` we don't have to change anything here.

Regarding the flow of events: even the document is not addressing to explain the flows, maybe https://gist.github.com/creydr/8b694956f27b0cb338b3576f3a893879#broker-trigger-context helps you understanding better the flows of the events. Each of the components (broker-ingress, channel&dispatcher, broker-filter) get an event with a token for THEIR audience. When sending it to the next step (e.g. ingress -> channel) a new JWT is requested for the new audience.

Does this make it clearer?

Reply:
That makes a lot of sense right now, thanks for the clarification! @creydr